cfme/containers/project.py

```diff
@@ -51,6 +52,31 @@ class ProjectCollection(BaseCollection):
 
     ENTITY = Project
 
+    def all(self):
+        # container_projects table has ems_id, join with ext_mgmgt_systems on id for provider name
+        project_table = self.appliance.db.client['container_projects']
+        ems_table = self.appliance.db.client['ext_management_systems']
+        provider = None
+        # filtered
+        if self.filters.get('provider'):
```

Comment:
I see that there is no difference between the queries instead of the `.filter` part - if that true - why not:
```python
        provider = self.filters.get('provider')
        project_query = (
                self.appliance.db.client.session
                    .query(project_table.name, ems_table.name)
                    .join(ems_table, project_table.ems_id == ems_table.id))
        if provider:
            project_query = project_query.filter(ems_table.name == provider.name))
```
I suggest the same for all the other objects.

Reply:
This is a great suggestion! I will implement it.

Reply:
Done